### ISSUE TITLE
MySQL 5.0.95 Replication Issue

### ISSUE THREAD

**Issue by temporaryaccount** [serge-rider/dbeaver#995]
We have 2 MySQL 5.0.95 servers with working master-master replication. The issue is that when working from DBeaver the replication does not work. I edit a value in one database, the other database doesn't get it. And vice-versa. When editing from another application (Navicat, mysql.exe, etc.) the replication works as intended. I presume this may be some kind of JDBC bug not related to DBeaver but investigate if you can.

P.S.
MariaDB 5.5 master-master replication with DBeaver's MariaDB driver does not have this issue. I haven't tried though to access the MySQL 5.0.95 servers using the MariaDB driver as I'm not sure whether it is safe.

**Comment by serge-rider** [comment]
Replication happens on server side - it can't be affected by any client drivers.
Try to switch to auto-commit mode. Maybe you don't see changes in replicated server because of transaction mode.

**Comment by temporaryaccount** [comment]
I understand it but the fact is the replication doesn't happen only when I use DBeaver. Auto-commit mode is enabled.

Maybe something legit which only DBeaver (or JDBC) is doing triggers some weird MySQL 5.0 bug. I edit a field in server 1, press Ctrl+s to save it, close the connection, open the server 2, the filed is in the old state. Close the connection, open server 1, my changes are there. I can try to investigate further (i.e. by analyzing MySQL's binary log files) if you are willing to hunt this bug down. Thanks.

**Comment by serge-rider** [comment]
Does this happen only if you edit data in data grid? What if you modify data with SQL queries?
(You can generate SQL script using action in data editor toolbar)

**Comment by temporaryaccount** [comment]
I just tried to modify data using SQL UPDATE query and the result is the same (no replication).

**Comment by serge-rider** [comment]
Maybe there is something in connection configuration (check that it is exactly the same as in other UI clients). I have no other ideas for now.

**Comment by babudro** [comment]
I am a new user of dBeaver -- trying to find a substitute for HeidiSQL using Wine in Linux -- and am also baffled by this behaviour with MariaDB 10.1.24.

If I alter a table, the change only appears on the master, it does not show up on the slaves.  I tail the mysql.log file and see that nothing is sent when dBeaver commits the change, although the master itself shows the change.

If I take the command shown in dBeaver's pop-up window before it commits a change and enter it from the command line it immediately shows in the log on the slaves as expected.

I opened dBeaver's transaction log and it shows nothing pending, and that the ALTER TABLE command completed with success.

I also tried disabling auto-commit, updating a record, and committing the change, but still nothing appeared at the replication slaves.  I even tried closing and re-opening dBeaver and still nothing appeared on the slaves; only on the master.

This seems a strange behaviour to do by default, if it is not a bug.

This behaviour has been observed with dBeaver Community edition 4.2.5 and 4.3.3.1, with Auto-Commit both enabled and disabled then manually committing the transactions.  No difference.
